openmeter/ent/schema/billing.go

```diff
@@ -0,0 +1,358 @@
+package schema
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+	"github.com/alpacahq/alpacadecimal"
+
+	"github.com/openmeterio/openmeter/openmeter/billing"
+	"github.com/openmeterio/openmeter/openmeter/billing/invoice"
+	"github.com/openmeterio/openmeter/openmeter/billing/provider"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type BillingProfile struct {
+	ent.Schema
+}
+
+func (BillingProfile) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+	}
+}
+
+func (BillingProfile) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("key").
+			NotEmpty().
+			Immutable(),
+		field.String("provider_config").
+			GoType(provider.Configuration{}).
+			ValueScanner(ProviderConfigValueScanner).
+			SchemaType(map[string]string{
+				"postgres": "jsonb",
```

Comment:
let's normalize config to enable migrations, indexing etc.

`workflow_auto_advance`, `workflow_due_after`, e.g.

Reply:
Added a new patch with the requested changes.